src/main/scala/uk/gov/hmrc/zap/api/ZapSetUp.scala

```diff
@@ -66,7 +66,15 @@ class ZapSetUp(zapClient: ZapClient) {
       callZapApi("/json/context/action/excludeFromContext", "contextName" -> zapContext.name, "regex" -> routeToBeIgnoredFromContext)
     }
   }
-}
 
+  def setConnectionTimeout(): Unit = {
+    val defaultTimeout: Int = 20
```

Comment:
There is a bit of duplication here as @dannyprok pointed out. We have a default value in the reference.conf and again here to check if the value in reference.conf is overwritten by the application.conf. We want to set the connectionTimeout only when the default value has changed. Is there a way to avoid this duplication?

Reply:
There is no need to set this deault. If you know that you set value in reference.conf you can just read from config and don't worry. Either your default or team's override will be used

Reply:
I was going to make a similar comment.  The recommended way to provide defaults with the typesafe config library is to provide them in the reference.conf: https://github.com/lightbend/config#how-to-handle-defaults.  The users of zap-automation will optionally provide their own `connectionTimeout` their application.conf, and if not provided, it will fallback to the default defined in reference.conf.  So I think there is no need to define a default again in `setConnectionTimeout`: simply use whatever was read in ZapConfig.

Reply:
This has be updated. We will now always make a call to setup connection timeout either with the default value in reference.conf or with the provided value.